src/components/DatePickerDecorator/DatePickerDecorator.js

```diff
@@ -0,0 +1,166 @@
+
+import { DATE_FORMAT } from '@folio/stripes-acq-components';
+
+import {
+  OptionsList,
+  TextDate,
+} from '../FlexibleForm/ControlDecorators/partials';
+import {
+  isFormattedMessage,
+  isTranslationId,
+  FORMS_SETTINGS,
+  ENTITY_KEYS,
+} from '../../utils';
+
+import styles from './DatePickerDecorator.css';
+
+export const DatePickerDecorator = memo(props => {
+  const {
+    id,
+    input,
+    wrapperLabel,
+    wrappedComponent,
+    ...rest
+  } = props;
+  const [currentValue, setCurrentValue] = useState(input?.value || '');
+  const [isDatepicker, setIsDatepicker] = useState(false);
+
+  const {
+    TODAY,
+    CHOOSE_DATE,
+  } = get(FORMS_SETTINGS, [ENTITY_KEYS.MAPPING_PROFILES, 'DECORATORS', 'DATE_PICKER'], []);
+
+  const dataOptions = [
+    {
+      value: TODAY.value,
+      name: <FormattedMessage id={TODAY.id} />,
+    },
+    {
+      value: CHOOSE_DATE.value,
+      name: <FormattedMessage id={CHOOSE_DATE.id} />,
+    },
+  ];
+
+  const handleChange = e => {
+    setCurrentValue(e.target ? e.target.value : e);
+    input.onChange(e);
+  };
+
+  const handleSetDate = (e, value) => {
+    const newValue = currentValue ? `${currentValue} "${value}"` : `"${value}"`;
+
+    setCurrentValue(newValue);
+    input.onChange(newValue);
+  };
+
+  const handleChangeWrapperValue = wrapperValue => {
+    let newValue = '';
+
+    if (wrapperValue === TODAY.value) {
+      setIsDatepicker(false);
+      newValue = currentValue ? `${currentValue} ###TODAY###` : '###TODAY###';
+      handleChange(newValue);
+    }
+
+    if (wrapperValue === CHOOSE_DATE.value) {
+      setIsDatepicker(true);
+    }
+  };
+
+  const needsTranslation = wrapperLabel && !isFormattedMessage(wrapperLabel) && isTranslationId(wrapperLabel);
+  const currentInput = useRef(input);
+  const Wrapper = !isDatepicker ? wrappedComponent : TextDate;
+
+  const {
+    onBlur,
+    onDragStart,
+    onDrop,
+    onFocus,
+  } = input;
+
+  let commonProps = {
+    value: currentValue,
+    inputRef: currentInput,
+    onBlur,
+    onChange: handleChange,
+    onDragStart,
+    onDrop,
+    onFocus,
+    ...rest,
+  };
+
+  const datePickerProps = {
+    onSetDate: handleSetDate,
+    dateFormat: DATE_FORMAT,
+  };
+
+  if (isDatepicker) {
+    commonProps = {
+      ...commonProps,
+      ...datePickerProps,
+    };
+  }
+
+  return (
+    <div
+      className={styles.decorator}
+      data-test-date-picker
+    >
+      <Wrapper
+        {...commonProps}
+      />
+      {needsTranslation ? (
+        <FormattedMessage id={wrapperLabel}>
+          {localized => (
+            <OptionsList
+              id={id}
+              label={localized}
+              dataOptions={dataOptions}
+              optionValue="value"
+              optionLabel="name"
+              className={styles['options-dropdown']}
+              onSelect={handleChangeWrapperValue}
+            />
+          )}
+        </FormattedMessage>
+      ) : (
+        <OptionsList
```

Comment:
I suggest implementing `WithTranslation` HOC which will decide under the hood whether the translation is needed so that the multiple instantiations which are kind of duplication would be eliminated. Under the hood I think it makes sense of using IntlConsumer and if `needsTransaltion` is true, then use `intl.formatMessage` and if false just provide the value.

```
<WithTranslation
  needsTransaltion={condition}
  value={wrapperLabel}
>
  {label => {
   <OptionsList
     ...
     label={label}
    ...

```

Reply:
Done